flexibleengine/resource_flexibleengine_dli_queues_v1.go

```diff
@@ -0,0 +1,244 @@
+)
+
+var regexp4Name = regexp.MustCompile(`^[a-z0-9_]+$`)
+
+func ResourceDliQueueV1() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceDliQueueCreate,
+		Read:   resourceDliQueueV1Read,
+		Delete: resourceDliQueueV1Delete,
+
+		Schema: map[string]*schema.Schema{
+			"queue_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+					v := val.(string)
+					if !regexp4Name.MatchString(v) {
+						errs = append(errs, fmt.Errorf("%q can contain only digits, lower letters, and underscores (_) ", key))
+					}
+					return
+				},
+			},
+
+			"queue_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "sql",
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"cu_count": {
+				Type:     schema.TypeInt,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"charging_mode": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+				Default:  1,
+			},
+
+			"enterprise_project_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "0",
+				StateFunc: func(val interface{}) string {
+					if val.(string) == "" {
+						return "0"
+					} else {
+						return val.(string)
+					}
+				},
+			},
+
+			"platform": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "x86_64",
+			},
+
+			"resource_mode": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"labels": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				ForceNew: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+
+			"feature": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "basic",
+			},
+
+			"tags": {
+				Type: schema.TypeMap,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"cidr_in_vpc": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"create_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceDliQueueCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	region := GetRegion(d, config)
+
+	dliClient, err := config.Config.DliV1Client(region)
+	if err != nil {
+		return fmt.Errorf("creating dli client failed: %s", err)
+	}
+
+	queueName := d.Get("queue_name").(string)
+
+	log.Printf("[DEBUG] create dli queues queueName: %s", queueName)
+	createOpts := queues.CreateOpts{
+		QueueName:           queueName,
+		QueueType:           d.Get("queue_type").(string),
+		Description:         d.Get("description").(string),
+		CuCount:             d.Get("cu_count").(int),
+		ChargingMode:        d.Get("charging_mode").(int),
```

Comment:
`ResourceMode` can be set to `0`. The type of `ResourceMode` in struct must be `*int`.
Otherwise, if the value is 0, this field will be ignored during JSON conversion.

Reply:
default value is 0, so delete the json tag :"omitempty"